volume_project/extract/2022/miovision_sqls/miovision_centreline_20220705.sql

```diff
@@ -0,0 +1,173 @@
+            ter.intersection_uid,
+            ter.lat,
+            ter.lng,
+            cent.geo_id,
+            ci.latitude,
+            ci.longitude
+        FROM miovision_api.intersections AS ter
+        LEFT JOIN gis.centreline_20220705 AS cent ON ter.int_id = cent.tnode
+        LEFT JOIN gis.centreline_intersection_20220705 AS ci ON cent.fnode = ci.int_id
+        WHERE cent.fcode_desc NOT IN (
+            'Trail',
+            'Geostatistical line',
+            'Other',
+            'River',
+            'Major Railway',
+            'Hydro Line',
+            'Walkway',
+            'Major Shoreline',
+            'Creek/Tributary',
+            'Ferry Route',
+            'Minor Railway',
+            'Minor Shoreline (Land locked)'
+        )
+
+        UNION ALL
+
+        SELECT DISTINCT
+            ter.intersection_uid,
+            ter.lat,
+            ter.lng,
+            cent.geo_id,
+            ci.latitude,
+            ci.longitude
+        FROM miovision_api.intersections AS ter
+        LEFT JOIN gis.centreline_20220705 AS cent ON ter.int_id = cent.fnode
+        LEFT JOIN gis.centreline_intersection_20220705 AS ci ON cent.tnode = ci.int_id
+        WHERE cent.fcode_desc NOT IN (
+            'Trail',
+            'Geostatistical line',
+            'Other',
+            'River',
+            'Major Railway',
+            'Hydro Line',
+            'Walkway',
+            'Major Shoreline',
+            'Creek/Tributary',
+            'Ferry Route',
+            'Minor Railway',
+            'Minor Shoreline (Land locked)'
+        )
+    ),
+
+    --determine difference in the "non-miovision node" co-ordinates - figure out if we're dealing with the north-south or east-west street in the intersection
+    ll_diff AS (
+        SELECT
+            ci.intersection_uid,
+            ci.lat,
+            ci.lng,
+            ci.geo_id,
+            ci.latitude,
+            ci.longitude,
+            ci.latitude - ci.lat AS lat_diff,
+            ci.longitude - ci.lng AS lng_diff,
+            CASE
+                WHEN abs(ci.latitude - ci.lat) > abs(ci.longitude - ci.lng) THEN 'NS'
+                WHEN abs(ci.longitude - ci.lng) > abs(ci.latitude - ci.lat) THEN 'EW'
+            END AS nsew
+        FROM cent_int AS ci
+    )
+
+    --determine which segments are N, S, E, W so that we can match with miovision leg + intersection_uid
+    SELECT
+        ld.geo_id AS centreline_id,
+        ld.intersection_uid,
+        CASE
+            WHEN ld.nsew = 'NS' AND ld.latitude > ld.lat THEN 'N'
+            WHEN ld.nsew = 'NS' AND ld.latitude < ld.lat THEN 'S'
+            WHEN ld.nsew = 'EW' AND ld.longitude > ld.lng THEN 'E'
+            WHEN ld.nsew = 'EW' AND ld.longitude < ld.lng THEN 'W'
+        END AS leg
+    FROM ll_diff AS ld
+);
+
+--2. It's important that the centreline to miovision file has the right combination of intersection_uids and legs. Here's how I checked that:
+WITH data_legs AS (
```

Comment:
@gabrielwol Would the intersection_movement table be interchangeable for this as well?

Reply:
Yeah that would probably give the same result a lot quicker!